pass/ProteusPass.cpp

```diff
@@ -868,6 +742,135 @@ class ProteusJitPassImpl {
       replaceWithJitLaunchKernel(M, CB);
   }
 
+  FunctionCallee getJitRegisterFatBinaryFn(Module &M) {
+    FunctionType *JitRegisterFatbinaryFnTy =
+        FunctionType::get(VoidTy, {PtrTy, PtrTy, PtrTy},
+                          /* isVarArg=*/false);
+    FunctionCallee JitRegisterFatbinaryFn = M.getOrInsertFunction(
+        "__jit_register_fatbinary", JitRegisterFatbinaryFnTy);
+
+    return JitRegisterFatbinaryFn;
+  }
+
+  void instrumentRegisterFatBinary(Module &M) {
+    Function *F = nullptr;
+
+    if (!RegisterFatBinaryName)
+      return;
+
+    F = M.getFunction(RegisterFatBinaryName);
+    if (!F)
+      return;
+
+    FunctionCallee JitRegisterFatBinaryFn = getJitRegisterFatBinaryFn(M);
```

Comment:
In the IRs that I am looking at this call is never being made when enabling rdc. It is only generated by the link.stub. I think you need access to the individual models (fatbins) as well.

Reply:
Proteus compiles the link.stub. I'll consider it resolved.